ps3.py

```diff
@@ -0,0 +1,346 @@
+    returns: dictionary (string -> int)
+    """
+    new_hand = hand.copy()
+    for litter in word.lower():
+        if litter in new_hand.keys() and new_hand[litter] > 0:
+            new_hand[litter] -= 1
+    return new_hand
+
+
+def is_valid_word(word, hand, word_list):
+    """
+    Returns True if word is in the word_list and is entirely
+    composed of letters in the hand. Otherwise, returns False.
+    Does not mutate hand or word_list.
+
+    word: string
+    hand: dictionary (string -> int)
+    word_list: list of lowercase strings
+    returns: boolean
+    """
+    word = word.lower()
+    if "*" in word:
+        for vowel in VOWELS:
+            if word.replace("*", vowel) in word_list:
+                return True
+        else:
+            return False
+    elif word in word_list:
+        for litter in word:
+            if (litter not in hand.keys()) or word.count(litter) > hand[litter]:
+                return False
+        else:
+            return True
+    else:
+        return False
+
+
+def calculate_handlen(hand):
+    """
+    Returns the length (number of letters) in the current hand.
+
+    hand: dictionary (string-> int)
+    returns: integer
+    """
+    count_letter = 0
+    for letter in hand.keys():
+        count_letter += hand[letter]
+    return count_letter
+
+
+def play_hand(hand, word_list, count_points):
+    """
+    Allows the user to play the given hand, as follows:
+
+    * The hand is displayed.
+
+    * The user may input a word.
+
+    * When any word is entered (valid or invalid), it uses up letters
+      from the hand.
+
+    * An invalid word is rejected, and a message is displayed asking
+      the user to choose another word.
+
+    * After every valid word: the score for that word is displayed,
+      the remaining letters in the hand are displayed, and the user
+      is asked to input another word.
+
+    * The sum of the word scores is displayed when the hand finishes.
+
+    * The hand finishes when there are no more unused letters.
+      The user can also finish playing the hand by inputing two
+      exclamation points (the string '!!') instead of a word.
+
+    hand: dictionary (string -> int)
+    word_list: list of lowercase strings
+    returns: the total score for the hand
+
+    """
+    if calculate_handlen(hand) != 0:
+        print("Current Hand:", end=" ")
+        display_hand(hand)
+        user_word = str(input("Enter word, or “!!” to indicate that you are finished:"))
+        if is_valid_word(user_word, hand, word_list):
+            count_points += get_word_score(user_word, calculate_handlen(hand))
+            print(user_word, "earned", str(get_word_score(user_word, calculate_handlen(hand))) + ".", " Total:",
+                  count_points, "points")
+            print(" ")
+        elif user_word == "!!":
+            return count_points
+        else:
+            print("This is not valid word. Please choose another word", "\n")
```

Comment:
There is no need in `\n` in the end, as print automatically move cursor to the next line in console.